ovos_persona_server/vector_stores.py

```
@@ -0,0 +1,676 @@
+
+        if content_bytes is None:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                                detail="File content not found for embedding.")
+
+        # Assuming text content for embedding. If binary, needs decoding or specific handling.
+        # For simplicity, assuming UTF-8 decodable text for chunking/embedding.
+        try:
+            text_content = content_bytes.decode('utf-8')
+        except UnicodeDecodeError:
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                                detail="File content is not valid UTF-8 text. Only text files are supported for embedding.") from e
```

Ruff (0.12.2)

Undefined name `e`

(F821)